hr_security_adj/models/hr_employee.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class Employee(models.Model):
+    _inherit = "hr.employee"
+
+    barcode = fields.Char(groups="hr.group_hr_user,hr_security_adj.group_hr_operator")
```

Comment:
Why did you add this part? Im not sure that background.

Reply:
In odoo15, there are two views, employee public and employee. For employee public, create and write options are blocked by view definition and it's model is hr.employee.public and use for public view. So, we need to use normal employee view and this views is restricted only for hr user(https://github.com/odoo/odoo/blob/8221bef0a023d33476bdaca868f2ff0991f12a0d/addons/hr/views/hr_views.xml#L18-L30) and barcode field is used as domain in this view. And barcode is allowed only for hr user. So, when we open with hr operator, we will get error. That's why we need to new operator group to barcode field.

Reply:
I see. Please add a comment so we don't get confused when we find it later.